### ISSUE TITLE
helm install with rbac.Enabled=false fails

### ISSUE THREAD

**Issue by Turil** [istio/istio#10519]
**Describe the bug**
When deploying Istio via Helm and Tiller, the config option global.rbacEnabled=false seems to be ignored. The command used was 
`helm install --set global.rbacEnabled=false install/kubernetes/helm/istio --name istio --namespace istio-system`
I tried to put --set global.rbacEnabled=false at the end of the command as well.

The output is

> Error: release istio failed: clusterroles.rbac.authorization.k8s.io "istio-galley-istio-system" is forbidden: attempt to grant extra privileges: [{[*] [admissionregistration.k8s.io] [validatingwebhookconfigurations] [] []} {[get] [config.istio.io] [*] [] []} {[list] [config.istio.io] [*] [] []} {[watch] [config.istio.io] [*] [] []} {[get] [*] [deployments] [istio-galley] []} {[get] [*] [endpoints] [istio-galley] []}] user=&{system:serviceaccount:kube-system:default c10d30ed-01f2-11e9-8d81-ca10d6d772c0 [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] map[]} ownerrules=[] ruleResolutionErrors=[clusterroles.rbac.authorization.k8s.io "system:discovery" not found]

Environment: Azure AKS cluster with 4 nodes, provisioned by Terraform

**Expected behavior**
istio should be installed despite the cluster not having rbac active

**Steps to reproduce the bug**
Create an Azure k8s cluster via Terraform. At the moment clusters created via AZ CLI or the Azure portal will have rbac active, but not via Terraform. The TF team is working on it (https://github.com/terraform-providers/terraform-provider-azurerm/pull/2495)

The underlying problem is, that in AKS the basic kubernetes roles do not get created at all when RBAC is disabled.

**Version**

> kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T20:17:28Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.5", GitCommit:"753b2dbc622f5cc417845f0ff8a77f539a4213ea", GitTreeState:"clean", BuildDate:"2018-11-26T14:31:35Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

**Installation**
Installation failed

**Environment**
Kubernetes cluster is running in AKS.

**Workaround info**
I managed to sneak in a (strange?) workaround:
I modified the helm-service-account.yaml in install/kubernetes/helm and added the following code to create the cluster-admin role:
```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  labels:
    kubernetes.io/bootstrapping: rbac-defaults
  name: cluster-admin
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
- nonResourceURLs:
  - '*'
  verbs:
  - '*'
---
```

After this the installation is successful but I'd like to avoid this extra / manual step for future deployments.

**Comment by linsun** [comment]
@mbanikazemi can you investigate this?  My gut feeling is many istio control plane component requires  RBAC so it is not easily to mark it as not enabled.

cc @esnible

**Comment by mbanikazemi** [comment]
I think the rbacEnabled option has no effect in the installation. It is not listed under available options for installation here: https://istio.io/docs/reference/config/installation-options/.
I believe it was introduced by the  #4738 (assorted fixes in pilot e2e tests) PR to enable/disable the rbac testing. 

However, the option is listed here: https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/README.md
This is confusing even though under the Prerequisites section it is stated that:
"Kubernetes 1.9 or newer cluster with RBAC (Role-Based Access Control) enabled is required".

In short, this flag is not to be used for installation without RBAC, and if this flag is used, it will not have any effect on the installation. Perhaps a better and less confusing name for the flag would have been rbacTestingEnabled.

**Comment by linsun** [comment]
Agreed on rename the flag to avoid the confusion.

**Comment by linsun** [comment]
@mbanikazemi can you submit a PR for this pls?

**Comment by linsun** [comment]
can we at least make the desp to be clear that this has nothing to do with the cluster has RBAC enabled or not?  I think everyone agreed this is confusing.  ideally, I'd like to remove it off the global config and not expose to any user.
